tools/ci_build/github/linux/run_build.sh

```diff
@@ -19,7 +19,7 @@ if [ $BUILD_DEVICE = "gpu" ]; then
         --config Debug Release \
         --skip_submodule_sync --enable_onnx_tests \
         --parallel --build_shared_lib \
-        --use_cuda --use_openmp \
+        --use_cuda --use_mkldnn --use_openmp \
```

Comment:
Some code is only reachable when mkldnn is not enabled. That's why I disabled it at here.  We already tested the mkldnn code in the CPU code, maybe we don't need to repeat it at there?

Reply:
The C# package for  GPU uses mkldnn during CPU execution -- so both are needed as execution providers for the GPU nuget package (because users should have a choice to use GPU or else CPU). If some code is unreachable with mkldnn enabled,  does the base non-GPU build have the same limitation, or only the GPU build?

Reply:
Is it possible for the c# package to make mkldnn provider as optional? If the onnxruntime.dll doesn't have this symbol, then the c# code won't need it. 

> If some code is unreachable with mkldnn enabled, does the base non-GPU build have the same limitation, or only the GPU build?
Both.

Reply:
In the current design, we can't.
The current design is we should allow people adding a new provider without changing any existing onnxruntime code. Therefore, there is no centralized place to list the providers.

Reply:
Removed mkldnn from this script. It'll still required for the GPU build however, but we can inject via yaml instead.